### ISSUE TITLE
Rebuild components in Entities received from server.

### ISSUE THREAD

**Issue by toms-dev** [toms-dev/typescript-game-engine-client#1]
Possibilities: 
- a type name is provided by the server, which is then mapped to a client class (extending Entity) that will setup the required components in it. => :+1: straightforward & quick implementation. :-1: not really flexible (the server can't add new/delete components on the go)
- a list of the client components to use is provided by the server => :no_entry:  induces coupling between server and client components
- a list of business-oriented "traits" is provided by the server and the client is able to rebuild the Entity dynamically and inserting the right components based on the local Trait definition. => :+1: loose coupling. This would allow dynamic modification of entities during gameplay (is this really needed? have to find a nice use case).
## About named types

There would be a map of type names to client _Entity_ classes (or builders) representing them.
The entity is then created and all its components are wired in its constructor for instance.
Some case could be problematic if one of the components needs a global service to work. (Possible solution: service locator pattern?)
## About the traits

The traits would be business-oriented definition of the entity, which means all the information should offer a level of abstraction that will concretized during the resolution performed by the Trait implementation.

Exemple: a Ship has: _Movement_, _Display_, _Shooting_, _Looting_, _Talking_.
Then the client would resolve the Display trait data as one or many given classes.
## Summing up

Named types should be fine to begin with, but it feels like the "trait" approach could be really interesting and offer more levels of abstraction, depending of what each developers wants. Basically, it would be possible to re-implement named types with the trait system, sending only one trait per Entity, gathering the list of all the needed components on client-side.
